tools/codegen/gen.py

```diff
@@ -1630,18 +1632,34 @@ def gen_op_headers(g: Union[NativeFunction, NativeFunctionsViewGroup]) -> List[s
 
         return {
             'ops_headers': gen_op_headers(g),
-            'func_definitions': list(concatMap(
-                lambda f: gen_functionalization_definition(
-                    selector,
-                    g,
-                    # We need to manually map inplace ops to their out-of-place variants
-                    # (we can't do this with NativeFunctionsGroup today because not all inplace ops have out= variants)
-                    None if isinstance(g, NativeFunctionsViewGroup) else to_functional_op.get(g.func.name, None)),
-                functions_needing_functionalization)),
-            'func_registrations': list(concatMap(
-                lambda f: gen_functionalization_registration(
-                    selector, f, backend_indices[DispatchKey.CompositeImplicitAutograd]),
-                functions_needing_functionalization)),
+            'func_definitions': gen_functionalization_definition(
+                selector,
+                g,
+                # We need to manually map inplace ops to their out-of-place variants
+                # (we can't do this with NativeFunctionsGroup today because not all inplace ops have out= variants)
+                None if isinstance(g, NativeFunctionsViewGroup) else to_functional_op.get(g.func.name, None),
+                FunctionalizationTarget.FUNCTIONALIZATION,
+            ),
+            'func_add_back_views_definitions': gen_functionalization_definition(
+                selector,
+                g,
+                # We need to manually map inplace ops to their out-of-place variants
+                # (we can't do this with NativeFunctionsGroup today because not all inplace ops have out= variants)
+                None if isinstance(g, NativeFunctionsViewGroup) else to_functional_op.get(g.func.name, None),
+                FunctionalizationTarget.ADD_BACK_VIEWS,
+            ),
+            'func_registrations': gen_functionalization_registration(
+                selector,
+                g,
+                backend_indices[DispatchKey.CompositeImplicitAutograd],
+                FunctionalizationTarget.FUNCTIONALIZATION,
+            ),
+            'func_add_back_views_registrations': gen_functionalization_registration(
+                selector,
+                g,
+                backend_indices[DispatchKey.CompositeImplicitAutograd],
+                FunctionalizationTarget.ADD_BACK_VIEWS,
+            ),
```

Comment:
It's not clear to me why you don't just generate both the addbackviews and regular things all in one go in a `gen_functionalization_registration` call. Save you a lot of typing here.

Reply:
(the thing Target exists for is when you're mapping over a list of functions, but no mapping is going on here...)